examples/ex24.cpp

```diff
@@ -0,0 +1,279 @@
+
+   // 3. Refine the mesh to increase the resolution. In this example we do
+   //    'ref_levels' of uniform refinement. We choose 'ref_levels' to be the
+   //    largest number that gives a final mesh with no more than 50,000
+   //    elements.
+   {
+      int ref_levels = (int)floor(log(50000./mesh->GetNE())/log(2.)/dim);
+      for (int l = 0; l < ref_levels; l++)
+      {
+         mesh->UniformRefinement();
+      }
+   }
+   mesh->ReorientTetMesh();
+
+   // 4. Define a parallel finite element space on the parallel mesh. Here we
+   //    use the Nedelec finite elements of the specified order.
+   FiniteElementCollection *fec = new ND_FECollection(order, dim);
+   FiniteElementCollection *H1fec = new H1_FECollection(order, dim);
+   FiniteElementSpace *fespace = new FiniteElementSpace(mesh, fec);
+   FiniteElementSpace *H1fespace = new FiniteElementSpace(mesh, H1fec);
+
+   int size = fespace->GetTrueVSize();
+   int H1size = H1fespace->GetTrueVSize();
+   cout << "Number of Nedelec finite element unknowns: " << size << endl;
+   cout << "Number of H1 finite element unknowns: " << H1size << endl;
+
+   // 5. Define the solution vector x as a parallel finite element grid function
+   //    corresponding to fespace. Initialize x by projecting the exact
+   //    solution. Note that only values from the boundary edges will be used
+   //    when eliminating the non-homogeneous boundary condition to modify the
+   //    r.h.s. vector b.
+   GridFunction x(fespace);
+   FunctionCoefficient p_coef(p_exact);
+   GridFunction p(H1fespace);
+   p.ProjectCoefficient(p_coef);
+
+   VectorFunctionCoefficient gradp_coef(sdim, gradp_exact);
+
+   // 6. Set up the bilinear forms.
+   Coefficient *muinv = new ConstantCoefficient(1.0);
+   Coefficient *sigma = new ConstantCoefficient(1.0);
+   BilinearForm *a = new BilinearForm(fespace);
+   MixedBilinearForm *a_NDH1 = new MixedBilinearForm(H1fespace, fespace);
+   if (pa)
+   {
+      a->SetAssemblyLevel(AssemblyLevel::PARTIAL);
+      a_NDH1->SetAssemblyLevel(AssemblyLevel::PARTIAL);
+   }
+
+   a->AddDomainIntegrator(new VectorFEMassIntegrator(*sigma));
+
+   a_NDH1->AddDomainIntegrator(new MixedVectorGradientIntegrator(*muinv));
+
+   // 7. Assemble the parallel bilinear form and the corresponding linear
+   //    system, applying any necessary transformations such as: parallel
+   //    assembly, eliminating boundary conditions, applying conforming
+   //    constraints for non-conforming AMR, static condensation, etc.
+   if (static_cond) { a->EnableStaticCondensation(); }
+
+   a->Assemble();
+   if (!pa) { a->Finalize(); }
+
+   a_NDH1->Assemble();
+   if (!pa) { a_NDH1->Finalize(); }
+
+   Vector B(fespace->GetTrueVSize());
+   Vector X(fespace->GetTrueVSize());
+
+   if (pa)
+   {
+      a_NDH1->Mult(p, x);
+      x.GetTrueDofs(B);
+   }
+   else
+   {
+      SparseMatrix& NDH1 = a_NDH1->SpMat();
+      Vector P(H1fespace->GetTrueVSize());
+      p.GetTrueDofs(P);
+
+      NDH1.Mult(P,B);
```

Comment:
I tried this with an AMR mesh (`./ex24 -m ../data/amr-quad.mesh`) and the code crashed here (in debug mode) since `P` is a T-vector but `NDH1` maps L-vector to L-vector, so the size of the input vector is incorrect.